subsys/mgmt/osdp/src/osdp_pd.c

```diff
@@ -432,53 +639,78 @@ static int pd_build_reply(struct osdp_pd *pd, uint8_t *buf, int max_len)
 		buf[len++] = BYTE_3(pd->id.firmware_version);
 		buf[len++] = BYTE_2(pd->id.firmware_version);
 		buf[len++] = BYTE_1(pd->id.firmware_version);
-		ret = 0;
+		ret = OSDP_PD_ERR_NONE;
 		break;
 	case REPLY_PDCAP:
-		if (max_len < REPLY_PDCAP_LEN) {
-			LOG_ERR(TAG "Out of buffer space!");
-			break;
-		}
+		ASSERT_BUF_LEN(REPLY_PDCAP_LEN);
 		buf[len++] = pd->reply_id;
-		for (i = 0; i < OSDP_PD_CAP_SENTINEL; i++) {
+		for (i = 1; i < OSDP_PD_CAP_SENTINEL; i++) {
 			if (pd->cap[i].function_code != i) {
 				continue;
 			}
 			if (max_len < REPLY_PDCAP_ENTITY_LEN) {
-				LOG_ERR(TAG "Out of buffer space!");
+				LOG_ERR("Out of buffer space!");
 				break;
 			}
 			buf[len++] = i;
 			buf[len++] = pd->cap[i].compliance_level;
 			buf[len++] = pd->cap[i].num_items;
 			max_len -= REPLY_PDCAP_ENTITY_LEN;
 		}
-		ret = 0;
+		ret = OSDP_PD_ERR_NONE;
 		break;
 	case REPLY_LSTATR:
-		if (max_len < REPLY_LSTATR_LEN) {
-			LOG_ERR(TAG "Out of buffer space!");
-			break;
-		}
+		ASSERT_BUF_LEN(REPLY_LSTATR_LEN);
 		buf[len++] = pd->reply_id;
 		buf[len++] = ISSET_FLAG(pd, PD_FLAG_TAMPER);
 		buf[len++] = ISSET_FLAG(pd, PD_FLAG_POWER);
-		ret = 0;
+		ret = OSDP_PD_ERR_NONE;
 		break;
 	case REPLY_RSTATR:
-		if (max_len < REPLY_RSTATR_LEN) {
-			LOG_ERR(TAG "Out of buffer space!");
-			break;
-		}
+		ASSERT_BUF_LEN(REPLY_RSTATR_LEN);
 		buf[len++] = pd->reply_id;
 		buf[len++] = ISSET_FLAG(pd, PD_FLAG_R_TAMPER);
```

Comment:
According to spec this status can take values in range: 0-2

Reply:
Right. But it is the current limitation of the implementation and this patch-set is not doing anything about it. We can take this up as a separate task once this PR is merged.